### ISSUE TITLE
Command Palette: consider bringing back more context to names of commands

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#54230]
## What problem does this address?

In testing out GB 16.6, I'm noticing that it's tricky to know which command to use for patterns, styles, pages as there's not much distinction in the duplicated items:

https://github.com/WordPress/gutenberg/assets/26996883/6a39ab60-27f8-4fb8-8d87-7c7251f493b3

Here's a specific example with Styles (of note, where is the style revision option?):

<img width="1335" alt="Screen Shot 2023-09-06 at 12 06 44 PM" src="https://github.com/WordPress/gutenberg/assets/26996883/738ccaed-ea16-4bc2-ac3d-b4a45c01bf85">

With just 6.3, while harder to skim, it's easier to get a sense of what each command will do:

https://github.com/WordPress/gutenberg/assets/26996883/fd8c78cc-1423-4de9-888a-01afa7a75f87

## What is your proposed solution?

I think we need to keep iterating on how verbose these descriptions are. Currently, I found I needed to guess which option might match what I'm trying to do between patterns, styles, and pages. cc @WordPress/gutenberg-design

**Comment by richtabor** [comment]
Yea, I agree. The tricky part is on these more generic locations, even some with the same icons. 

I'm not sure why "Open style revisions" is now just "Styles" though — that doesn't seem right. 

Here's an issue on adding context to post types: https://github.com/WordPress/gutenberg/issues/51505

**Comment by richtabor** [comment]
And templates should be clearly identified as templates:
 
<img width="418" alt="CleanShot 2023-09-06 at 15 48 37" src="https://github.com/WordPress/gutenberg/assets/1813435/1bb6a8c2-9b52-4508-ab51-8b44b2683bb5">

Why I leaned towards this exploration in https://github.com/WordPress/gutenberg/issues/53192#issuecomment-1660561371, where the category/type/source could be exposed like "Templates > Index": 

<img width="620" alt="image" src="https://github.com/WordPress/gutenberg/assets/1813435/475fc830-e1de-45c5-89fc-d4f859f5af64">

**Comment by jameskoster** [comment]
I know we initially decided against this, but I still think the command palette could benefit from more organisation patterns in order for it to scale.

<img width="1416" alt="commandpalette" src="https://github.com/WordPress/gutenberg/assets/846565/500677d0-1829-4af0-850c-3e7ac4657547">

**Comment by annezazu** [comment]
Yeah, I am feeling that latter design might be helpful to explore at this stage.

**Comment by richtabor** [comment]
Or even just the context on the right side.

**Comment by richtabor** [comment]
What about if we just append the context to the title? I.e. "Index template" instead of "Index"?

<img width="499" alt="CleanShot 2023-09-11 at 19 19 33" src="https://github.com/WordPress/gutenberg/assets/1813435/86b562d4-d512-4d46-aa7f-ddd8692ad7cf">

**Comment by t-hamano** [comment]
> What about if we just append the context to the title? I.e. "Index template" instead of "Index"?

I think this issue needs to be considered for localization. Depending on the language, the word order may not always be `{template-part-name} template part` or `{pattern-name} pattern`. Words may be reversed, words may be hyphenated, and words may not be lowercased.

The following is an example of a pattern description.

### English (en)

`Header pattern`

![en](https://github.com/WordPress/gutenberg/assets/54422211/b871c074-4082-4a1a-b5e7-5403afb01a66)

### French (fr_FR)

`composition Header`

![fr_FR](https://github.com/WordPress/gutenberg/assets/54422211/cef83999-c497-452d-a9d2-6c916f212feb)

### German (de_DE)

`Header-Vorlage`

![de_DE](https://github.com/WordPress/gutenberg/assets/54422211/d7444ecb-340a-44ae-bb8d-0d2329c0466c)

### Italiano (it_IT)

`pattern Header`

![it_IT](https://github.com/WordPress/gutenberg/assets/54422211/4a106e51-7ed6-4e45-ae31-1352ce0cdf21)

**Comment by jameskoster** [comment]
Additionally it wouldn't work so well for other post types "Hello World post" is a bit strange. It would produce a lot of repetition too. Groups with headings solves those issues. The headings could even drill down into contextual views:

<img width="1674" alt="commandpalette" src="https://github.com/WordPress/gutenberg/assets/846565/1c09f49a-e0af-43d5-9d04-1c7a2ab59a59">

**Comment by richtabor** [comment]
> I think this issue needs to be considered for localization. 

Yea, I'll push @youknowriad's suggestion [here](https://github.com/WordPress/gutenberg/pull/54366/files/bda70a10378cb749b58e01ed375995c696f008e9#r1322644712).

**Comment by richtabor** [comment]
> Groups with headings solves those issues. 

I think we can solve it without having to split up the command list, keeping the flow predictable and simple. I don't personally mind appended template/template part. 

But also the text on the right side could be helpful (https://github.com/WordPress/gutenberg/issues/53192), especially if folks start adding third-party commands. It's an extra affordance to help identify what the command is/where it's from, rather than if it's a command or shortcut.  

For templates, we could have "Templates" as the extra affordance.

**Comment by jameskoster** [comment]
Yeah there's probably many ways to slice it. 

> It's an extra affordance to help identify what the command is/where it's from, rather than if it's a command or shortcut.

Might be worth trying this. Having the identifier in a consistent position seems way more scannable compared to the suffix. Makes the localisation element simpler too.

**Comment by jameskoster** [comment]
@richtabor unsure if this is what you meant, but it could work:

<img width="1416" alt="commandpalette2" src="https://github.com/WordPress/gutenberg/assets/846565/7325faf4-3b39-41c1-bd1f-9c7accdc6bc8">

It solves all the issues with the suffix approach, and plays nicely with the third-party identification pattern you proposed.

**Comment by richtabor** [comment]
> @richtabor unsure if this is what you meant, but it could work:

Yea I think so.

**Comment by annezazu** [comment]
More food for thought with the Command Palette from the [FSE Outreach Program's Final Touches call for testing:](https://make.wordpress.org/test/2023/09/18/fse-program-testing-call-26-final-touches/#comment-3177)

>It would be awesome to have more information about the current state of a toggle, “Disable distraction free mode” instead of “Toggle distraction free mode.” Or maybe make the icon of the active toggle reflect its state by adding a black background like the Settings or Styles buttons do.

[& more feedback:](https://make.wordpress.org/test/2023/09/18/fse-program-testing-call-26-final-touches/#comment-3178)

>Using the Command Palette. I write Portfolio and get up two Portfolios that have two different icons. One that looks like a post/page and the other as if it is a template part. Hovering over both it would be nice if it said something like: Portfolio page and the other Portfolio template part. To strengthen the meaning of the icons I see.

**Comment by danielbachhuber** [comment]
On a WordPress multisite install, it would be nice to have an affordance for selecting a site to run the command on.

**Comment by SaxonF** [comment]
Grouping is worth revisiting. `cmdk` has `<Command.Group heading="Letters">`. This works quite well for other products like Linear and Vercel.

https://github.com/WordPress/gutenberg/assets/1072756/de232ef9-7a69-4db8-9577-8e103267120e

https://github.com/WordPress/gutenberg/assets/1072756/589709f3-044c-4ccb-844f-4767809d4fd0

You can also see nested commands which would resolve scenarios like @danielbachhuber multisite or digging into a post type commands without navigating.

**Comment by fcoveram** [comment]
The design introduced in #75616 addresses the main suggestion and most, if not all, ideas shared later. I would close this one and open a new one if needed. Especially if there is an ongoing call for feedback.

**Comment by jasmussen** [comment]
Sounds good, closing in favour of #75616 and any new issues.
